### PR TITLE
Add MM32 SDK and USB driver

### PR DESCRIPTION
**Pull request by zhangslice**
Signed-off-by: zhangslice <1304224508@qq.com>

**Describe the PR**
A clear and concise description of what this PR solve.

**Additional context**
If applicable, add any other context about the PR and/or screenshots here.

### CODE REVIEW

**Review by hathach — COMMENTED** [review]
the hw/mcu should be added as submodule, do mindmotion has github repo for mm322 sdk. else I will help and make one and push update to this PR for you guys.

**Comment by zhangslice** [comment]
The following repository can be used, and the SDK will be updated if there are any updates or additions
https://github.com/zhangslice/mm32sdk


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "hathach/tinyusb"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月2日(星期三) 中午12:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [hathach/tinyusb] Add MM32 SDK and USB driver (#869)





 
@hathach commented on this pull request.
 
the hw/mcu should be added as submodule, do mindmotion has github repo for mm322 sdk. else I will help and make one and push update to this PR for you guys.

**Comment by hathach** [comment]
> The following repository can be used, and the SDK will be updated if there are any updates or additions https://github.com/zhangslice/mm32sdk

thanks that repo would be useful, I notice that both Keil_core and IAR_Core within the cmsis is the same, without any file differences. you should just have it as `CSMSIS/Core` https://github.com/zhangslice/mm32sdk/tree/main/mm32f327x/CMSIS

do you want to merge them together ? I am fine If you still want to keep it that way though.

**Comment by hathach** [comment]
Ok thanks for letting me, in general you should not modify those file since they are from ARM which already taken care of the keil/gcc/iar. But that is up to you.

**Comment by zhangslice** [comment]
Well, thank you, that's a good suggestion




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "hathach/tinyusb"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月2日(星期三) 下午2:24
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [hathach/tinyusb] Add MM32 SDK and USB driver (#869)





 
Ok thanks for letting me, in general you should not modify those file since they are from ARM which already taken care of the keil/gcc/iar. But that is up to you.

File: src/portable/mm32/dcd_mm32f327x_otg.c [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+/// @attention
+///
+/// THE EXISTING FIRMWARE IS ONLY FOR REFERENCE, WHICH IS DESIGNED TO PROVIDE
+/// CUSTOMERS WITH CODING INFORMATION ABOUT THEIR PRODUCTS SO THEY CAN SAVE
+/// TIME. THEREFORE, MINDMOTION SHALL NOT BE LIABLE FOR ANY DIRECT, INDIRECT OR
+/// CONSEQUENTIAL DAMAGES ABOUT ANY CLAIMS ARISING OUT OF THE CONTENT OF SUCH
+/// HARDWARE AND/OR THE USE OF THE CODING INFORMATION CONTAINED HEREIN IN
+/// CONNECTION WITH PRODUCTS MADE BY CUSTOMERS.
+///
+/// <H2><CENTER>&COPY; COPYRIGHT MINDMOTION </CENTER></H2>
```
Comment (hathach) [inline src/portable/mm32/dcd_mm32f327x_otg.c:15]: copyright for the drivers file that is within `/src` must be MIT license to be compatible with the rest of the project. Please consult your company and put the MIT license here if you are willing to move forward with the PR.

**Review by hathach — CHANGES_REQUESTED** [review]
I pushed the update to your master branch on your fork, please pull before making any other commit and update. There are 2 requests for change as follows:
- MIT license is required for `dcd_mm32f327x_otg.c`
- `tinyusb_keil.zip` must be removed

File: src/portable/mm32/dcd_mm32f327x_otg.c (hunk 1 above)
Comment (hathach) [reply src/portable/mm32/dcd_mm32f327x_otg.c:15]: Note: only this file need to be MIT license. Other driver in your sdk does not need to change at all

**Review by hathach — APPROVED** [review]
The port still need bsp to verify that mcu work with existing project on mm32. However, it is good as intial PR.

Note: I have removed the keil project .zip, such as binary file is not allowed in this repo. You could probably provide it as additional, add-on to your sdk.

**Pull request merged**